crates/bevy_pbr/src/render/mesh.rs

```diff
@@ -1117,6 +1317,16 @@ pub fn extract_meshes_for_gpu_building(
             );
         },
     );
+
+    // Also record info about each mesh that became invisible.
+    let mut queue = render_mesh_instance_queues.borrow_local_mut();
+    for entity in removed_visibilities_query
+        .read()
+        .chain(removed_global_transforms_query.read())
+        .chain(removed_meshes_query.read())
+    {
+        queue.remove(entity.into(), any_gpu_culling);
+    }
```

Comment:
This needs to check whether the entity has had the components re-added after they were removed.
Currently, adding the following system to an example will cause the meshes to disappear and re-appear every other frame:
```rust
.add_systems(
    Update,
    |query: Query<Entity, With<Mesh3d>>, mut commands: Commands| {
        for entity in &query {
            commands
                .entity(entity)
                .remove::<GlobalTransform>()
                .insert(GlobalTransform::default());
        }
    },
)
```

Reply:
Good catch, I pushed a fix.

Reply:
I've verified the fix :+1: